compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -322,3 +322,11 @@ fromAnyContractKey (AnyContractKey any rep)
 
 deriving instance Eq Archive
 deriving instance Show Archive
+
+-- | True if contract exists, fails if current context doesn't have permission to know.
+canSeeContractWithKey : forall t k. (HasLookupByKey t k) => k -> Update Bool
```

Comment:
I'm not the biggest fan of this name but can't really think of anything better. Ideally, this would end in `ByKey`, like the other primitives for contract keys. How about `doesExistByKey`? cc @bame-da @cocreature

Reply:
But this is not checking for existence, as it may exist without you having permission to see it.

Reply:
I agree that the name sounds a bit weird but I cannot come up with anything better either. I do agree with @garyverhaegen-da that `doesExistByKey` is not an improvement.

Reply:
@bame-da Do you have something to add to the bikeshedding I initiated?

Reply:
What about `canFetchContractByKey`?

Reply:
That sounds pretty confusing given that it deliberately has different auth rules than `fetchByKey`.

Reply:
How about `existsKey`. It will almost always be used like

```
e <- existsKey @t key
if e
  then ... 
  else ...
```

With that name I can read that as "If exists key k then ... else ..."

Reply:
I like this one as well, even though it's not about the existence of the key, but the contract with that key.

Reply:
> That sounds pretty confusing given that it deliberately has different auth rules than `fetchByKey`.

I don't think so, given that `fetchByKey` explodes. It seems reasonable to me that it takes more care to ensure it will not blow up before calling it than to just call it and hope for the best.

Also, the name makes sense to me, given that the whole point of this function is to tell you whether you can call `fetchByKey` without exploding.

Reply:
Having `canFetchContractByKey` return `False` in a case where `fetchByKey` succeeds seems pretty confusing to me. I think you need at least something like `canSafelyFetch…` but that’s an awkward name.

Reply:
Updated to `visibleByKey` for now as that seems to be the favourite per the ongoing Slack discussion. I'll give it one more day to settle down, and for anyone who really dislikes it to reach out.